docs/Design/SoftDetailedDes/MIS.tex

```diff
@@ -242,6 +242,93 @@ \subsubsection{Local Functions}
 
 \newpage
 
+\section{MIS of Recommendation Module} \label{mRecommendation}
+
+Module for recommending user purchases.
+
+\subsection{Module}
+
+recommendationModule
+
+\subsection{Uses}
+
+Database Driver Module (Section \ref{mDBDriver})
+
+\subsection{Syntax}
+
+\subsubsection{Exported Constants}
+
+None
+
+\subsubsection{Exported Access Programs}
+
+\begin{center}
+    \begin{tabular}{p{5cm} p{4cm} p{3cm} p{4cm}}
+    \hline
+    \textbf{Name} & \textbf{In} & \textbf{Out} & \textbf{Exceptions} \\
+    \hline
+    collectItemData & Item & - & InvalidDataFormatException \\
+    getCheaperAlternatives & itemId & List$<$itemId$>$ & - \\
+    getRecommendationDetails & itemId & Item & - \\
```

Comment:
Once again, can you add a sentence at the beginning referencing where the definition of Item is, i.e. {mExtraction}?

Furthermore, itemId isn't a type. It's of type Integer I think if you're referencing the database type, so instead of itemId, it should be $\mathbb{Z}$.